### PR TITLE
To be able to sort icons

### PR DESCRIPTION
**Pull request by nfxpnk**
The idea is to name icons as 001-star.svg, 002-dot.svg, 003-plus.svg to be able to sort icons, because I need to display all icons in my styleguide sorted by types (for an example all arrows, all social media icons, etc.)

All i need is to name icon correctly with number and then just substr icon-id

PR it is just for a reference. I think it's better to have an option to be able to turn on custom sorting

### CODE REVIEW

**Comment by tancredi** [comment]
Hi @nfxpnk - Firstly, thank you for the PR!!! 🙏 

While I think there's value in the ability of ordering icons, there's also been a proposal on the Discord channel of implementing a feature to allow prefixing icon IDs with code-points:

> What would you think about getting more control of the codepoints in Fantasticon? So you would have a folder with the name <00.regular> and a folder with the name <e000.pua> and if you put 0063_questionmark.svg in the regular folder it will actually map to the ASCII codepoint of a question mark. It would also mean that any codepoints will stay the same even when we add other svg's, and we could then also use [data-icon]:before{content: attr(data-icon)}, because no matter what the user does, the glyph is assigned to that codepoint. Using data-icon in HTML then also means that we do not need an enormous list of partly never-used classes in our CSS.

I think both are valuable, but I struggle to think of a way I can make them co-exist or pick.

I'm a bit worried here because it seems like either are use-cases that not everyone may be interested in and may grow the codebase in complexity, especially if supported together, will need documenting etc..

Also had a small play with the PR and currently naming conventions support directories name-spacing, so if you have lots of icons of different sorts you can arrange them like such:

```
./social
   ./facebook.svg
   ./twitter.svg
./glyph
    ./chevron
        ./down.svg
        ./up.svg
```

And the generated icon IDs would be:

`["social-facebook", "social-facebook", "glyph-chevron-up", "glyph-chevron-down"]`

(This is a feature I should take the time to document better on the readme, and lacked time for)

This PR currently breaks with this implementation, so it would probably need to work on both the file base-name and the path segments.

Basically, at the minute I'd need a little bit more urgency this use-case before starting adding logic to ignore part of the filenames arbitrarily. Does this make sense?

**Comment by nfxpnk** [comment]
Sure, thanks!

**Comment by saiballo** [comment]
I achieved my goal with a node module used as a wrapper. substantially when I run my script:

    I read the json map generated by fantasticon
    I read all *svg files in input folder and compare with the previous json list to remove some entries
    Create new codepoints file based on json map (replace \\ with 0x)
    Passing codepoints file to cli command of fantasticonr with parameter --codepoints

with these steps my font map never change even I update icons folder

**Comment by tancredi** [comment]
This is now achievable using a custom `getIconId` method - you can find more details in the readme.

Closing this, hope it addresses your issue nicely! :)
